File: leet_2006_count_pairs_with_k_diff.cpp
class Solution {
public:
    int countKDifference(vector<int>& nums, int k)
    {
        int r = 0;
        int len = nums.size(), i = -1, j;

        while (++i < len)
        {
            j = i;
            while (++j < len)
            {
                if (abs(nums[i] - nums[j]) == k)
                    r++;
            }
        }
        return r;
    }
};

File: leet_2006_count_pairs_with_k_diff.py
class Solution:
    def countKDifference(self, n: List[int], k: int) -> int:
        r = 0
        for i in range(len(n)):
            for j in range(i + 1, len(n)):
                if abs(n[i] - n[j]) == k:
                    r += 1
        return r 
